docs/protocol_v1_suggestion.md

```diff
@@ -0,0 +1,383 @@
+}
+```
+
+#### Allocations
+
+Track fund availability and net balance per token:
+```solidity
+struct Allocation {
+    address token;          // Token address (address(0) for native tokens)
+    uint256 available;      // Amount available for use
+    uint256 locked;         // Amount locked/reserved
+    uint256 netValue;       // Absolute value of (deposits - withdrawals)
+    bool isNetPositive;     // True if (deposits - withdrawals) >= 0
+}
+```
+
+**Multi-asset Support**: The channel supports multiple tokens by having one allocation per token in the allocations array.
+
+The allocation structure maintains:
+- **available**: Funds that can be freely used in state transitions
+- **locked**: Funds that are reserved/locked for specific purposes
+- **netValue + isNetPositive**: Tracks the net flow (deposits minus withdrawals), which can be negative
+
+**Note**: `available + locked` may differ from `netValue` due to cross-chain transfers:
+- After sending funds: `available` decreases but `netValue` remains unchanged
+- After receiving funds: `available` increases but `netValue` remains unchanged
+- Only deposit/withdraw operations affect `netValue`
+
+### Channel Lifecycle States
+
+```solidity
+enum ChannelStatus {
+    VOID,     // Channel doesn't exist
+    ACTIVE,   // Channel is operational
+    DISPUTE,  // In challenge period
+    FINAL     // Closed (ephemeral state)
+}
+```
+
+### Fund Management
+
+The protocol maintains two separate fund pools:
+
+1. **Custody Ledger**: User's deposited funds that can be withdrawn directly
+2. **Custody Liquidity**: Contract's liquidity pool that changes only during state updates (cannot be withdrawn directly)
+
+#### State Update Rules
+
+For each allocation in a state update, the following rules are enforced:
+
+1. **Net Value Changes**:
+   - If `netValueDiff = newNetValue - previousNetValue > 0`: Move `netValueDiff` from custody ledger to custody liquidity
+   - If `netValueDiff = newNetValue - previousNetValue < 0`: Move `|netValueDiff|` from custody liquidity to custody ledger
+
+2. **Available Balance Changes**:
+   - If `availableDiff = newAvailable - previousAvailable > 0`: Subtract `availableDiff` from custody liquidity
+   - If `availableDiff = newAvailable - previousAvailable < 0`: Add `|availableDiff|` to custody liquidity
+
+3. **Locked Balance Changes**:
+   - If `lockedDiff = newLocked - previousLocked > 0`: Add `lockedDiff` to custody liquidity
+   - If `lockedDiff = newLocked - previousLocked < 0`: Subtract `lockedDiff` from custody liquidity
+
+## Protocol Workflows
+
+### 1. Channel Creation
+
+**Steps:**
+
+```
+create(Channel, State) → channelId
+```
+- Anyone can submit the transaction
+- Initial state must have both user and broker signatures
+- Channel immediately becomes ACTIVE
+
+**Cross-Chain Architecture**:
+- Each channel is unique to its chain (channelId includes chainId)
+- Broker coordinates related channels across different chains
```

Comment:
What do you mean by "related channels"? Do you mean "channels of the same user"?